Automattic/wp-calypso
# Update/update ad trackers

### In this this PR:

_This is a continuation of this [PR](https://github.com/Automattic/wp-calypso/pull/33525)._

- **Google Analytics**: 
  - New Gtag implementation.
  - Legacy e-commerce plugin removed. If we decide we need it we'll be using the new Gtag based one.
  - Removed the separate `google-analytics` switch, now it is enabled by default with all the rest when `ad-tracking` is `true`.
- **Google Ads**: 
  - Now WordPress.com and Jetpack use the same MCC pixel, removed the separate Jetpack Gtag.
  - Removed legacy "AdWords" pixel and events.
- **Facebook**: 
  - Both WordPress.com and Jetpack pixels receive Purchase and AddToCart events.
  - `initFacebook` and `initFacebookAdvancedMatching` are now one function.
- **Pinterest**: update ID to `2613194105266` and fix event order issue.
- **LinkedIn**: disabled.
- **Twitter**: disabled.
- **Nanigans**: removed.
- Added new GA and Tracks events `calypso_new_user_site_creation` and removed `is_new_user` fuzziness.
- Completely removed `recordProduct()` function which was firing conversions on a per-cart-item basis (deprecated legacy behaviour).

### Testing

- MAKE SURE I HAVEN'T REVERTED THIS COMMIT: https://github.com/Automattic/wp-calypso/pull/34116/files

- Enable logs: `localStorage.setItem( 'debug', 'calypso:analytics:*' );`

- Click on cookie banner or set `document.cookie = 'sensitive_pixel_option=yes;'`

- Reload http://calypso.localhost:3000/?flags=gdpr-banner,ad-tracking

- Alternatively, enable `gdpr-banner` and `ad-tracking` in `config/development.json` and restart Calypso.

- You should see `calypso:analytics:utils isAdTrackingAllowed: true` in your JS console.

- Make sure Google Analytics works as expected.
- Make sure Facebook pixel works as expected.
- Make sure `calypso_new_user_site_creation` is fired as expected when a new user creates a new site.
- Make sure all key events work as expected (Registration, Signup, AddToCart, Purchase, PageView)
- Make sure all enabled trackers still work.
- Make sure there are no errors.
